api/v1/controllers/CheckInController.js

```diff
@@ -0,0 +1,80 @@
+var bodyParser = require('body-parser');
+var _Promise = require('bluebird');
+
+var services = require('../services');
+var middleware = require('../middleware');
+var requests = require('../requests');
+var roles = require('../utils/roles');
+var errors = require('../errors');
+
+var router = require('express').Router();
+
+
+function updateCheckInByUserId (req, res, next) {
+    services.CheckInService
+        .findCheckInByUserId(req.params.id)
+        .then(function (checkin){
+            return services.CheckInService.updateCheckIn(checkin, req.body);
+        })
+        .then(function (response){
+            res.body = response.toJSON();
+            return next();
+        })
+        .catch(function (error){
+            return next(error);
+        });
+}
+
+function fetchCheckInByUserId (req, res, next) {
+    services.CheckInService
+        .findCheckInByUserId(req.params.id)
+        .then(function (checkin){
+            res.body = checkin.toJSON();
+            return next();
+        })
+        .catch(function (error){
+            return next(error);
+        });
+}
+
+function fetchCheckInByUser (req, res, next) {
+    services.CheckInService
+        .findCheckInByUserId(req.user.id)
+        .then(function (checkin){
+            res.body = checkin.toJSON();
+            return next();
+        })
+        .catch(function (error){
+            return next(error);
+        });
+}
+
+function createCheckIn (req, res, next) {
+    services.CheckInService
+        .createCheckIn(req.params.id, req.body)
+        .then(function (checkin){
+            // console.log(checkin);
+            res.body = checkin.toJSON();
+            return next();
+        })
+        .catch(function (error){
+            return next(error);
+        })
+}
+
+
+router.use(bodyParser.json());
+router.use(middleware.auth);
+
+router.post('/:id', middleware.request(requests.CheckInRequest),
```

Comment:
The POST here should not have an `:id` URL parameter. Instead, the request body should specify a the `userId`. This likely means that you need two separate types of `CheckInRequests` or other special handling because the `userId` should not be changed by a PUT.